feat: automate slack sns subscription

JustinDPerkins opened the pull request:
## Cloud One Service
[ X] **File Storage Security**




## Proposed Changes
Python script to take Slack plugin lambda and subscribe it to all deployed FSS storage stacks.

JustinDPerkins commented:
@raphabot I went the extra mile and added another step for further instruction on that.

felipecosta09 commented on File-Storage-Security/Post-Scan-Action-Automations/aws-slack-notification-automation/ReadMe.md:
```diff
@@ -0,0 +1,32 @@
+# Subscribe deployed FSS Storage Stack SNS topics to Slack Notification Plugin.
```
Small comment: 
- Keep the file nomeclature as `README.md` just to keep the standard with the rest of the repo.

JustinDPerkins replied:
fixed

felipecosta09 commented on File-Storage-Security/Post-Scan-Action-Automations/aws-slack-notification-automation/ReadMe.md:
````diff
@@ -0,0 +1,32 @@
+  
+  * Obtain the following parameters.
+      - API Key - Generate a [Cloud One API Key](https://cloudone.trendmicro.com/docs/account-and-user-management/c1-api-key/)
+      - Cloud One Account Region - example: ```us-1```.
````
Will be helpful to our people overseas to reference all the regions with this link -> https://cloudone.trendmicro.com/docs/identity-and-account-management/c1-regions/

So they don't need to find the other regions themselves.

JustinDPerkins replied:
Added that in README

felipecosta09 commented on File-Storage-Security/Post-Scan-Action-Automations/aws-slack-notification-automation/python-aws-automate-slack-subscription.py:
```diff
@@ -0,0 +1,67 @@
+        for output in outputs:
+            keyName = output["OutputKey"]
+            if keyName == "ScanResultTopicARN":
+                # print(output["OutputValue"])
```
Should we remove this test?

JustinDPerkins replied:
Removed. Was just for testing.

felipecosta09 commented on File-Storage-Security/Post-Scan-Action-Automations/aws-slack-notification-automation/python-aws-automate-slack-subscription.py:
```diff
@@ -0,0 +1,67 @@
+        stack_names.append(name['name'])
+    describe_fss_storage(stack_names)
+
+
```
The only function without a comment 😢

JustinDPerkins replied:
No functions shall lack comments.

felipecosta09 reviewed (changes requested):
Very small changes to the README and python code overall.

felipecosta09 commented:
A general comments about the PATH chosen for the project... Don't you think that this automation fit in the "Deployment" folder instead of being a "Post-Scan-Action-Automations"?

I just thinking about the next ones that might come up, `Deployment` is more generic, might be more logical to a person looking at these automations in the repo...

Thoughts @raphabot @JustinDPerkins ?

JustinDPerkins commented:
@felipecosta09 I moved the folder structure under Deployment. At least with a directory called 'post-scan-action-automations', it's kind of easy telling. Always happy to change it

JustinDPerkins commented:
@raphabot and @felipecosta09 
I updated the script to leverage an AWS profile param. 
I updated the ReadMe to reflect changes to call the script.
I added in a reqs file as well.
Last is now this script lets sns topics in other AWS regions to be subscribed.

felipecosta09 reviewed (dismissed):
LGTM

felipecosta09 commented:
> LGTM.
> 
> However I'd recommend to make the profile parameter non-mandatory as it also isn't in the AWS CLI. I'd default to the default profile.

Thoughts before we merge @JustinDPerkins ?

JustinDPerkins commented:
I added a diagram to this to assist in script usage.
I changed the awsprofile flag to not be required and will default to 'default' if not defined.
Added instruction link to set up named profile.
Also this will work for other FSS notification plugins not just slack(IE MS Teams, Jira, etc). So I updated the ReadMe to be more agnostic.

felipecosta09 reviewed (approved):
LGTM

JustinDPerkins commented:
New logo inserted.
